src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -0,0 +1,71 @@
+package tech.cryptonomic.conseil.metadata
+
+import tech.cryptonomic.conseil.config.MetadataOverridesConfiguration
+import tech.cryptonomic.conseil.config.Platforms.PlatformsConfiguration
+import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, Entity, Network, Platform}
+import tech.cryptonomic.conseil.metadata.Data.{AttributeData, EntityData, NetworkData, PlatformData}
+import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
+import tech.cryptonomic.conseil.util.OptionUtil.when
+import tech.cryptonomic.conseil.util.ConfigUtil
+
+import scala.concurrent.Future.successful
+import scala.concurrent.{ExecutionContext, Future}
+
+class MetadataService(config: PlatformsConfiguration,
```

Comment:
I find the code in this class a little complicated, and I'm trying to figure out if 
- we could easily separate the override logic from the configuration reading
- the nesting of different maps/flatMaps might be simplified

I'll forego a deeper analysis now, as long as there will be tests to guarantee correctness of the code.
If some idea to simplify comes along, I'll be happy with that, but for now it's good to go for me, as long as it works.

Reply:
I refactored it a little bit. Hope you find it more clear now. I spend a lot of time on it. Mostly because there is a lot of types like ``Future[Option[List[...]]]``.

Reply:
Did you consider using some of cats' MonadT? Or `Nested`?

Reply:
Doesn't look bad at all, 👍